File: Sem2/PO_projekt/c++/animals/sheep.cpp
#include "sheep.hpp"

Sheep::Sheep(int x, int y, Map* map)
    : Animal(x, y, 4, 4, map, "Sheep") {};

Sheep* Sheep::reproduce(int newX, int newY) {
    return new Sheep(newX, newY, map);
}

File: Sem2/PO_projekt/Java/animals/Sheep.java
package Java.animals;
import Java.Animal;
import Java.Organism;
import Java.Game;
import java.awt.Color;

public class Sheep extends Animal {
    public Sheep(int strength, int initiative, int x, int y, Game game) {
        super(strength, initiative, x, y, game);
    }

    @Override
    public Color getColor() {
        return new Color(255, 255, 255); // white
    }

    @Override
    public Organism getOrganism() {
        return this;
    }

    @Override
    protected Animal reproduce(int x, int y) {
        return new Sheep(4, 4, x, y, game);
    }

}
